sdk/go/common/workspace/plugins.go

```diff
@@ -1812,19 +1812,42 @@ func getPluginInfoAndPath(
 						(stat.Mode()&0o100 != 0 || runtime.GOOS == windowsGOOS) {
 						logging.V(6).Infof("GetPluginPath(%s, %s, %v): found next to current executable %s",
 							kind, name, version, candidate)
-
-						return &PluginInfo{
-							Kind: kind,
-							Name: name,
-							Path: filepath.Dir(candidate),
-						}, candidate, nil
+						bundledPath = candidate
+						break
 					}
 				}
 			}
 		}
 	}
 
-	// Otherwise, check the plugin cache.
+	// We prefer the ambient path, but we need to check if this is the same as the bundled
+	// path to decide if we're warning or not.
+	pluginPath := bundledPath
+	if ambientPath != "" {
+		if ambientPath != bundledPath {
+			// They don't match _but_ it might be they just don't match because the pulumi install is symlinked,
+			// e.g. /opt/homebrew/bin/pulumi-language-nodejs -> /opt/homebrew/Cellar/pulumi/3.77.0/bin/pulumi-language-nodejs
+			// So before we warn, lets just check if we can resolve symlinks in the ambient path and then check again.
+			fullAmbientPath, err := filepath.EvalSymlinks(ambientPath)
```

Comment:
Instead of special-casing here, it might be preferable to always EvalSymlinks up above when we calculate ambientPath originally.

```go
	var ambientPath string
	if includeAmbient {
		path, err := exec.LookPath(filename)
		if err == nil {
			path, err = filepath.EvalSymlinks(path)
		}
		if err == nil {
			ambientPath = path
		}
```

e.g., the bundledPath already gets an EvalSymlinks because of this:

```go
		exePath, exeErr := os.Executable()
		if exeErr == nil {
			fullPath, fullErr := filepath.EvalSymlinks(exePath)
```

Reply:
Yeh I wasn't confident about how these paths are used that changing them in that way wouldn't have any effect.
I _think_ it would be fine but was trying to make this change as unobservable as possible just in case.

Reply:
Ah, fair. Yeah, in that case I can see the value of doing that just here.